crates/hyperswitch_connectors/src/connectors/nexixpay/transformers.rs

```diff
@@ -90,50 +415,201 @@ impl TryFrom<&ConnectorAuthType> for NexixpayAuthType {
+pub enum NexixpayRefundOperationType {
+    Refund,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
+pub enum NexixpayRefundResultStatus {
+    Pending,
+    Voided,
+    Refunded,
     Failed,
-    #[default]
-    Processing,
+    Executed,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct NexixpayTransactionResponse {
+    order_id: String,
+    operation_id: String,
+    operation_result: NexixpayPaymentStatus,
+    operation_type: NexixpayOperationType,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct NexixpayRSyncResponse {
+    order_id: String,
+    operation_id: String,
+    operation_result: NexixpayRefundResultStatus,
+    operation_type: NexixpayRefundOperationType,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct NexixpayPaymentsCaptureRequest {
+    amount: StringMinorUnit,
+    currency: Currency,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct NexixpayPaymentsCancelRequest {
+    description: Option<String>,
+    amount: StringMinorUnit,
+    currency: Currency,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct NexixpayOperationResponse {
+    operation_id: String,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct NexixpayRefundRequest {
+    pub amount: StringMinorUnit,
+    pub currency: Currency,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct RefundResponse {
+    operation_id: String,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct NexixpayErrorBody {
+    pub code: Option<String>,
+    pub description: Option<String>,
+}
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct NexixpayErrorResponse {
+    pub errors: Vec<NexixpayErrorBody>,
 }
 
-impl From<NexixpayPaymentStatus> for common_enums::AttemptStatus {
+impl From<NexixpayPaymentStatus> for AttemptStatus {
     fn from(item: NexixpayPaymentStatus) -> Self {
         match item {
-            NexixpayPaymentStatus::Succeeded => Self::Charged,
-            NexixpayPaymentStatus::Failed => Self::Failure,
-            NexixpayPaymentStatus::Processing => Self::Authorizing,
+            NexixpayPaymentStatus::Declined
+            | NexixpayPaymentStatus::DeniedByRisk
+            | NexixpayPaymentStatus::ThreedsFailed
+            | NexixpayPaymentStatus::Failed => Self::Failure,
+            NexixpayPaymentStatus::Authorized | NexixpayPaymentStatus::ThreedsValidated => {
+                Self::Authorized
+            }
+            NexixpayPaymentStatus::Executed => Self::Charged,
+            NexixpayPaymentStatus::Pending => Self::AuthenticationPending,
```

Comment:
Is it always `authentication_pending`?, this will be translated to the merchant as requires_customer_action

Reply:
as this is called only from authorize and psync calls this will not cause any issue.

Reply:
Please add a comment here why we are hardcoding status.